components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/api/client/GradleEnterpriseApiClient.java

```diff
@@ -258,6 +258,22 @@ private String avoidanceOutcome(GradleBuildCachePerformanceTaskExecutionEntry ta
         return avoidanceOutcome.toString();
     }
 
+    private Duration buildTimeFrom(GradleBuildCachePerformance buildCachePerformance) {
+        return toDuration(buildCachePerformance.getBuildTime());
+    }
+
+    private Duration buildTimeFrom(MavenBuildCachePerformance buildCachePerformance) {
+        return toDuration(buildCachePerformance.getBuildTime());
+    }
+
+    private BigDecimal serializationFactorFrom(GradleBuildCachePerformance buildCachePerformance) {
+        return BigDecimal.valueOf(buildCachePerformance.getSerializationFactor());
+    }
+
+    private BigDecimal serializationFactorFrom(MavenBuildCachePerformance buildCachePerformance) {
+        return BigDecimal.valueOf(buildCachePerformance.getSerializationFactor());
+    }
+
```

Comment:
I would move these 4 methods to right below `summarizeTaskExecutions` to have a consistent order with how they are invoked further up.